test/integration/narrow/routes/vet/sheep-epg.test.js

```diff
@@ -0,0 +1,97 @@
+const cheerio = require('cheerio')
+const getCrumbs = require('../../../../utils/get-crumbs')
+const expectPhaseBanner = require('../../../../utils/phase-banner-expect')
+
+describe('Sheep EPG  test', () => {
+  const auth = { credentials: { reference: '1111', sbi: '111111111' }, strategy: 'cookie' }
+  const url = '/vet/sheep-epg'
+
+  describe(`GET ${url} route`, () => {
+    test('returns 200', async () => {
+      const options = {
+        method: 'GET',
+        url,
+        auth
+      }
+
+      const res = await global.__SERVER__.inject(options)
+
+      expect(res.statusCode).toBe(200)
+      const $ = cheerio.load(res.payload)
+      expect($('h1').text()).toMatch('Percentage reduction in eggs per gram (EPG) from pre- to post- worming treatment')
+      expectPhaseBanner.ok($)
+    })
+
+    test('when not logged in redirects to /vet', async () => {
+      const options = {
+        method: 'GET',
+        url
+      }
+
+      const res = await global.__SERVER__.inject(options)
+
+      expect(res.statusCode).toBe(302)
+      expect(res.headers.location).toEqual('/vet')
+    })
+  })
+
+  describe(`POST ${url} route`, () => {
+    let crumb
+    const method = 'POST'
+
+    beforeEach(async () => {
+      crumb = await getCrumbs(global.__SERVER__)
+    })
+
+    test.each([
+      { epg: 50 },
+      { epg: 2 }
+    ])('returns 302 to next page when acceptable answer given', async ({ epg }) => {
+      const options = {
+        method,
+        url,
+        payload: { crumb, epg },
+        auth,
+        headers: { cookie: `crumb=${crumb}` }
+      }
+
+      const res = await global.__SERVER__.inject(options)
+
+      expect(res.statusCode).toBe(302)
+      expect(res.headers.location).toEqual('/vet/declaration')
+    })
+
+    test.each([
+      { epg: 200, message: 'Error: "epg" must be less than or equal to 100' },
+      { epg: -80, message: 'Error: "epg" must be greater than or equal to 0' }
+    ])('returns error when unacceptable answer is given', async ({ epg, message }) => {
```

Comment:
I don't think these are the errors you want? They aren't the bespoke ones you have added.

In the validation code the key used is `string.<thing>` but the type from Joi is number. If you update the keys to be `number.<thing>` then you'll get the error messages you want as opposed to those default ones.

If you change the objects used to have keys for `number.base`, `number.min` and `number.max`. Then you can have an error for `Enter EPG percentage`, `EPG percentage must be 0 or more` and `EPG percentage must be 100 or less` respectively.

i.e.
```js
.messages({
  'number.base': epgErrorMessages.validEpg,
  'number.max': 'EPG percentage must be 100 or less',
  'number.min': 'EPG percentage must be 0 or more'
})

```